server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -282,7 +343,12 @@ void ensureRepositorySet() {
         }
         final String remoteStoreRepo = REMOTE_CLUSTER_STATE_REPOSITORY_SETTING.get(settings);
         assert remoteStoreRepo != null : "Remote Cluster State repository is not configured";
-        final Repository repository = repositoriesService.get().repository(remoteStoreRepo);
+        final Repository repository;
+        try {
+            repository = repositoriesService.get().repository(remoteStoreRepo);
+        } catch (RepositoryMissingException e) {
+            return;
```

Comment:
Seems like this defeats the purpose of a method called "ensure repository set", no?

Reply:
yes, added it for testing, forgot to remove. will update.